lib/asset_sync/config.rb

```diff
@@ -222,11 +233,20 @@ def fog_options
           :google_storage_secret_access_key => google_storage_secret_access_key,
           :google_storage_access_key_id => google_storage_access_key_id
         })
+      elsif azure_rm?
+        require 'fog/azurerm'
+        options.merge!({
+          :azure_storage_account_name => azure_storage_account_name,
+          :azure_storage_access_key   => azure_storage_access_key,
+        })
+        options.merge!({
+          :environment => fog_region
+        }) if fog_region
       else
         raise ArgumentError, "AssetSync Unknown provider: #{fog_provider} only AWS, Rackspace and Google are supported currently."
       end
 
-      options.merge!({:region => fog_region}) if fog_region && !rackspace?
+      options.merge!({:region => fog_region}) if fog_region && !rackspace? && !azure_rm?
```

Comment:
Should really put this line into every `if` except `rackspace`

Reply:
I can refactor this later you can leave this unchanged :P

Reply:
Do you mean like below?

```ruby
if aws?
  ...
  options.merge!({:region => fog_region}) if fog_region
elsif rackspace?
  ...
  options.merge!({:rackspace_region => fog_region}) if fog_region
elsif google?
  ...
  options.merge!({:region => fog_region}) if fog_region
elsif azure_rm?
  ...
  options.merge!({:environment => fog_region}) if fog_region
else
  ...
end
```